chains/evm/listener/event-handler.go

```diff
@@ -29,27 +28,28 @@ func NewETHEventHandler(address common.Address, client ChainClient) *ETHEventHan
 	}
 }
 
-func (e *ETHEventHandler) HandleEvent(sourceID, destID uint8, depositNonce uint64, rID [32]byte) (*relayer.Message, error) {
-	addr, err := e.matchResourceIDToHandlerAddress(rID)
+func (e *ETHEventHandler) HandleEvent(sourceID, destID uint8, depositNonce uint64, resourceID [32]byte, calldata, handlerResponse []byte) (*relayer.Message, error) {
+	handlerAddr, err := e.matchResourceIDToHandlerAddress(resourceID)
 	if err != nil {
 		return nil, err
 	}
 
-	eventHandler, err := e.matchAddressWithHandlerFunc(addr)
+	eventHandler, err := e.matchAddressWithHandlerFunc(handlerAddr)
 	if err != nil {
 		return nil, err
 	}
 
-	return eventHandler(sourceID, destID, depositNonce, addr, e.client)
+	return eventHandler(sourceID, destID, depositNonce, resourceID, calldata, handlerResponse)
 }
 
-func (e *ETHEventHandler) matchResourceIDToHandlerAddress(rID [32]byte) (common.Address, error) {
+// matchResourceIDToHandlerAddress is a private method that matches a previously registered resource ID to its corresponding handler address
+func (e *ETHEventHandler) matchResourceIDToHandlerAddress(resourceID [32]byte) (common.Address, error) {
 	definition := "[{\"inputs\":[{\"internalType\":\"bytes32\",\"name\":\"\",\"type\":\"bytes32\"}],\"name\":\"_resourceIDToHandlerAddress\",\"outputs\":[{\"internalType\":\"address\",\"name\":\"\",\"type\":\"address\"}],\"stateMutability\":\"view\",\"type\":\"function\"}]"
```

Comment:
We should probably change now all ABIs to get definition from consts (or in separate PR)